src/stdlib_utils/parallelism_framework.py

```diff
@@ -234,6 +246,10 @@ def _sleep_for_idle_time_during_iteration(
             self._idle_iteration_time_ns += idle_time_ns
             time.sleep(idle_time_ns / 10 ** 9)
 
+    @staticmethod
```

Comment:
is there a reason this is a static method instead of just a general (private) function in the module scope? do we anticipate objects inheriting this class needing to use it or some other reason?